src/util/PathUtil.ts

```diff
@@ -5,6 +5,45 @@ import type { ResourceIdentifier } from '../http/representation/ResourceIdentifi
 import type { HttpRequest } from '../server/HttpRequest';
 import { BadRequestHttpError } from './errors/BadRequestHttpError';
 
+// Characters to ignore when URL decoding the URI path to a file path.
+const pathComponentDelimiters = [ '/', '\\' ];
+// Regex to find all instances of encoded path component delimiters.
+const encodedDelimiterRegex = new RegExp(
+  `${pathComponentDelimiters.map((item): string => encodeURIComponent(item)).join('|')}`, 'giu',
```

Comment:
I'd do

```suggestion
  `${pathComponentDelimiters.map((delimiter): string => encodeURIComponent(item)).join('|')}`, 'giu',
```

throughout (especially with the arrays below at 17 and 22) but that's a crazy nit